src/Cli/dotnet/Installer/Windows/WindowsUtils.cs

```diff
@@ -54,16 +55,18 @@ public static bool InstallRunning()
         }
 
         /// <summary>
-        /// Queries the Windows Update Agent API to determine if there is a pending reboot.
+        /// Queries the Windows Update Agent, Component Based Servicing (CBS), and pending file rename registry keys to determine if there is a pending reboot.
         /// </summary>
         /// <returns><see langword="true"/> if there is a pending reboot; <see langword="false"> otherwise.</see></returns>
-        /// <remarks>
-        /// See <see href="https://docs.microsoft.com/en-us/windows/win32/api/wuapi/nf-wuapi-isysteminformation-get_rebootrequired">this</see>
-        /// for more information.
-        /// </remarks>
         public static bool RebootRequired()
         {
-            return new SystemInformationClass().RebootRequired;
+            using RegistryKey auKey = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired");
+            using RegistryKey cbsKey = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending");
+            using RegistryKey sessionKey = Registry.LocalMachine.OpenSubKey(@"SYSTEM\CurrentControlSet\Control\Session Manager");
```

Comment:
This will not work properly for a 32-bit dotnet.exe running on a 64-bit OS.  You need to use the 64-bit registry view:
```suggestion
            using RegistryKey localMachineKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry64);
            using RegistryKey auKey = localMachineKey.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired");
            using RegistryKey cbsKey = localMachineKey.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending");
            using RegistryKey sessionKey = localMachineKey.OpenSubKey(@"SYSTEM\CurrentControlSet\Control\Session Manager");
```

Reply:
Thanks, I completely blanked on the redirection. I've created a mop-up PR